torch/distributions/multivariate_normal.py

```diff
@@ -90,36 +100,44 @@ class MultivariateNormal(Distribution):
         scale_tril (Tensor): lower-triangular factor of covariance, with positive-valued diagonal
 
     Note:
-        Only one of `covariance_matrix` or `scale_tril` can be specified.
+        Only one of :attr:`covariance_matrix` or :attr:`precision_matrix` or
+        :attr:`scale_tril` can be specified.
 
-        Using `scale_tril` will be more efficient: all computations internally
-        are based on `scale_tril`. If `covariance_matrix` is passed instead,
-        this is only used to compute the corresponding lower triangular
-        matrices using a Cholesky decomposition.
+        Using :attr:`scale_tril` will be more efficient: all computations internally
+        are based on :attr:`scale_tril`. If :attr:`covariance_matrix` or
+        :attr:`precision_matrix` is passed instead, it is only used to compute
+        the corresponding lower triangular matrices using a Cholesky decomposition.
     """
     arg_constraints = {'loc': constraints.real_vector,
                        'covariance_matrix': constraints.positive_definite,
                        'scale_tril': constraints.lower_cholesky}
     support = constraints.real
     has_rsample = True
 
-    def __init__(self, loc, covariance_matrix=None, scale_tril=None, validate_args=None):
+    def __init__(self, loc, covariance_matrix=None, precision_matrix=None, scale_tril=None, validate_args=None):
         event_shape = torch.Size(loc.shape[-1:])
-        if (covariance_matrix is None) == (scale_tril is None):
-            raise ValueError("Exactly one of covariance_matrix or scale_tril may be specified (but not both).")
-        if scale_tril is None:
+        if (covariance_matrix is not None) + (scale_tril is not None) + (precision_matrix is not None) != 1:
+            raise ValueError("Exactly one of covariance_matrix or precision_matrix or scale_tril may be specified.")
+        if scale_tril is not None:
+            if scale_tril.dim() < 2:
+                raise ValueError("scale_tril matrix must be two-dimensional, with optional leading batch dimensions")
+            self.scale_tril = scale_tril
+            batch_shape = _get_batch_shape(scale_tril, loc)
+        elif covariance_matrix is not None:
             if covariance_matrix.dim() < 2:
                 raise ValueError("covariance_matrix must be two-dimensional, with optional leading batch dimensions")
             self.covariance_matrix = covariance_matrix
             batch_shape = _get_batch_shape(covariance_matrix, loc)
         else:
-            if scale_tril.dim() < 2:
-                raise ValueError("scale_tril matrix must be two-dimensional, with optional leading batch dimensions")
-            self.scale_tril = scale_tril
-            batch_shape = _get_batch_shape(scale_tril, loc)
+            if precision_matrix.dim() < 2:
+                raise ValueError("precision_matrix must be two-dimensional, with optional leading batch dimensions")
```

Comment:
Shouldn't this be at least two-dimensional?

Reply:
Yes. This line raises error if it is less than two dimensional.

Reply:
I thought it would be better to print the error message as `precision_matrix must be at least two-dimensional, ....`

Reply:
Oh I see, and I agree. I just copied the original error message for cov. I'll update those as well. :)

Reply:
Updated.

Reply:
Thank you. :)